data/data/bootstrap/files/usr/local/bin/installer-gather.sh

```diff
@@ -0,0 +1,107 @@
+for container in $(podman ps --all --quiet)
+do
+    podman logs "${container}" >& "${ARTIFACTS}/bootstrap/containers/${container}.log"
+    podman inspect "${container}" >& "${ARTIFACTS}/bootstrap/containers/${container}.inspect"
+done
+
+# Collect cluster data
+function queue() {
+    local TARGET="${ARTIFACTS}/${1}"
+    shift
+    local LIVE="$(jobs | wc -l)"
+    while [[ "${LIVE}" -ge 45 ]]; do
+    sleep 1
+    LIVE="$(jobs | wc -l)"
+    done
+    # echo "${@}"
+    if [[ -n "${FILTER}" ]]; then
+    "${@}" | "${FILTER}" >"${TARGET}" &
+    else
+    "${@}" >"${TARGET}" &
+    fi
+}
+OC="oc --config=/etc/kubernetes/kubeconfig --insecure-skip-tls-verify --request-timeout=5s"
+
+mkdir -p "${ARTIFACTS}/control-plane" "${ARTIFACTS}/workers" "${ARTIFACTS}/resources/pods" "${ARTIFACTS}/resources/network" "${ARTIFACTS}/resources/nodes"
+
+echo "Gathering cluster resources ..."
+queue resources/nodes.list ${OC} get nodes -o jsonpath --template '{range .items[*]}{.metadata.name}{"\n"}{end}'
+queue resources/containers ${OC} get pods --all-namespaces --template '{{ range .items }}{{ $name := .metadata.name }}{{ $ns := .metadata.namespace }}{{ range .spec.containers }}-n {{ $ns }} {{ $name }} -c {{ .name }}{{ "\n" }}{{ end }}{{ range .spec.initContainers }}-n {{ $ns }} {{ $name }} -c {{ .name }}{{ "\n" }}{{ end }}{{ end }}'
+queue resources/api-pods ${OC} get pods -l openshift.io/component=api --all-namespaces --template '{{ range .items }}-n {{ .metadata.namespace }} {{ .metadata.name }}{{ "\n" }}{{ end }}'
+
+queue resources/apiservices.json ${OC} get apiservices -o json
+queue resources/clusteroperators.json ${OC} get clusteroperators -o json
+queue resources/clusterversion.json ${OC} get clusterversion -o json
+queue resources/configmaps.json ${OC} get configmaps --all-namespaces -o json
+queue resources/csr.json ${OC} get csr -o json
+queue resources/endpoints.json ${OC} get endpoints --all-namespaces -o json
+queue resources/events.json ${OC} get events --all-namespaces -o json
+queue resources/kubeapiserver.json ${OC} get kubeapiserver -o json
+queue resources/kubecontrollermanager.json ${OC} get kubecontrollermanager -o json
+queue resources/machineconfigpools.json ${OC} get machineconfigpools -o json
+queue resources/machineconfigs.json ${OC} get machineconfigs -o json
+queue resources/namespaces.json ${OC} get namespaces -o json
+queue resources/nodes.json ${OC} get nodes -o json
+queue resources/openshiftapiserver.json ${OC} get openshiftapiserver -o json
+queue resources/pods.json ${OC} get pods --all-namespaces -o json
+queue resources/rolebindings.json ${OC} get rolebindings --all-namespaces -o json
+queue resources/roles.json ${OC} get roles --all-namespaces -o json
+queue resources/secrets.json ${OC} get secrets --all-namespaces -o json
+queue resources/services.json ${OC} get services --all-namespaces -o json
+
+FILTER=gzip queue resources/openapi.json.gz ${OC} get --raw /openapi/v2
+
+echo "Gathering node journals ..."
```

Comment:
There is an expectation that the apiserver is not up when we are running this script.

Reply:
Right, added a TODO item to handle that, but I don't think it would hurt to attempt to collect those (along with other resources which assume API server is working)

Reply:
Some APIs may be available even if bootstrap failed. I believe @deads2k called out he'd like to fetch a list of api resources if possible.

Reply:
> There is an expectation that the apiserver is not up when we are running this script.

This script must not die if the KAS is down and it should using the KAS wherever it can.  That means that @abhinavdahiya is correct in saying that you cannot rely on `oc adm node-logs` to work.

I think that a best effort section for gathering cluster-specific resources that does a series of `oc get pod -ojson`, `oc get node -ojson` and the like (see https://github.com/openshift/release/blob/master/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L361-L381) is valuable.

Reply:
:+1: on not relying on `oc adm node-logs` we need a 100% successful method for that.